cvmfs/catalog_mgr_impl.h

```diff
@@ -826,11 +827,19 @@ bool AbstractCatalogManager<CatalogT>::MountSubtree(
        iEnd = nested_catalogs.end(); i != iEnd; ++i)
   {
     // Next nesting level
-    PathString nested_path_slash(i->mountpoint);
-    nested_path_slash.Append("/", 1);
-    if (path_slash.StartsWith(nested_path_slash)) {
+    if (path.StartsWith(i->mountpoint)) {
+      // in this case the path doesn't start with 
+      // the mountpoint in a file path sense
+      // (e.g. path is /a/bc and mountpoint is /a/b), and will be ignored
+      //
+      // checking it this way allows us to avoid the 
```

Comment:
I think this last sentence of the comment is not clear without seeing the diff with the previous version of the code. Let's remove it.